shipment_planning_maxoptra/wizard/shipment_maxoptra_schedule_import.py

```diff
@@ -121,44 +122,33 @@ def regroup_operations(
-                previous_moves = picking_moves.move_orig_ids
-                pickings = previous_moves.picking_id
-                for _loc, previous_pickings in groupby(
-                    pickings, key=lambda m: m.location_id
-                ):
-                    for pick in previous_pickings:
-                        # Avoid rescheduling same picking multiple times
-                        if pick.id in scheduled_picking_ids:
-                            continue
-                        new_batch = self.env["stock.picking.batch"].create({})
-                        new_batch_picking_ids.append(new_batch.id)
-                        pick_values = {"batch_id": new_batch.id}
-                        # TODO: add field in wizard to allow for pick
-                        # planning simuntaneously or successively.
-                        # For more information see ROADMAP.rst
-                        if start_datetime and operation_duration:
-                            hours, minutes = format_duration(operation_duration).split(
-                                ":"
-                            )
-                            delay = relativedelta(
-                                hours=cnt * int(hours), minutes=cnt * int(minutes)
-                            )
-                            pick_values["scheduled_date"] = start_datetime + delay
-                        pick.write(pick_values)
-                        scheduled_picking_ids.append(pick.id)
-                        # increment counter manually as we cannot use enumerate
-                        #  on the two nested loops
-                        cnt += 1
-        return self.env["stock.picking.batch"].browse(new_batch_picking_ids)
+        batch_obj = self.env["stock.picking.batch"]
+        cnt = 0
+        # TODO check JIT flow for picks sorted("scheduled_date", reverse_order)
+        original_pickings = following_batch_pickings.picking_ids
+        previous_pickings = original_pickings.move_lines.move_orig_ids.picking_id
+        for _type, pick_list in self._group_to_batch_flow(previous_pickings):
+            picks = self.env["stock.picking"].browse([pick.id for pick in pick_list])
+            new_batch = batch_obj.create({"picking_ids": [(6, 0, picks.ids)]})
+            new_batch_picking_ids.append(new_batch.id)
+            if start_datetime and operation_duration:
+                # TODO pickings should be loaded to track in particular order
+                for pick in picks.sorted("scheduled_date", reverse_order):
+                    hours, minutes = format_duration(operation_duration).split(":")
+                    delay = relativedelta(
+                        hours=cnt * int(hours), minutes=cnt * int(minutes)
+                    )
+
+                    # TODO define is start_datime global for all pickings in the wizard
+                    # or separate for each batch
+                    pick.write({"scheduled_date": start_datetime + delay})
```

Comment:
To me there's an issue here.

As I understand, we want multiple pickings on the same batch with different scheduled dates? Then I don't really get what's the point of using a batch if they are not supposed to be processed together.

Then all the "previous operations" will share the same picking type and we'll have only a single batch but with different scheduled dates, so IMO we need to introduce another configuration option instead of changing the actual way. Could be done by changing `regroup_operations` from a Boolean to a Selection field.

I'm sorry but I have a hard time understanding this change :confused: